specification/azsadmin/resource-manager/compute/Microsoft.Compute.Admin/preview/2018-07-30-preview/Disks.json

```diff
@@ -0,0 +1,319 @@
+  },
+  "host": "adminmanagement.local.azurestack.external",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "paths": {
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Compute.Admin/locations/{location}/admindisks": {
+      "get": {
+        "x-ms-examples": {
+          "Returns a list of disks.": {
+            "$ref": "./examples/Disks/List.json"
+          }
+        },
+        "tags": [
+          "Disks"
+        ],
+        "operationId": "Disks_List",
+        "description": "Returns a list of disks.",
+        "parameters": [
+          {
+            "$ref": "../2015-12-01-preview/Compute.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../2015-12-01-preview/Compute.json#/parameters/LocationParameter"
+          },
+          {
+            "$ref": "Disks.json#/parameters/TenantSubscriptionIdParameter"
+          },
+          {
+            "$ref": "Disks.json#/parameters/DiskStateParameter"
+          },
+          {
+            "$ref": "Disks.json#/parameters/SourceSharePathParameter"
+          },
+          {
+            "$ref": "Disks.json#/parameters/DiskCountParameter"
+          },
+          {
+            "$ref": "Disks.json#/parameters/DiskStartParameter"
+          },
+          {
+            "$ref": "Disks.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK -- The list of disks has been returned.",
+            "schema": {
+              "$ref": "Disks.json#/definitions/DiskList"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": null
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Compute.Admin/locations/{location}/admindisks/{DiskId}": {
+      "get": {
+        "x-ms-examples": {
+          "Returns the requested disk.": {
+            "$ref": "./examples/Disks/Get.json"
+          }
+        },
+        "tags": [
+          "Disks"
+        ],
+        "operationId": "Disks_Get",
+        "description": "Returns the disk.",
+        "parameters": [
+          {
+            "$ref": "../2015-12-01-preview/Compute.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../2015-12-01-preview/Compute.json#/parameters/LocationParameter"
+          },
+          {
+            "$ref": "Disks.json#/parameters/DiskIdParameter"
+          },
+          {
+            "$ref": "Disks.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK -- The disk has been returned.",
+            "schema": {
+              "$ref": "Disks.json#/definitions/Disk"
+            }
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "DiskProperties": {
```

Comment:
I see that you have readOnly: false on multiple properties on this object.  Could you verify that making them mutable makes sense on all of them?   For instance I see sharePath is mutable, but that seems like something that should not be modified.

Reply:
read-only property will not be serialized to server side. Our rest api takes "Disks" as input objects. And we don't have to any assumption about all the information is right from customer. So all those information is able to change and serialize to server. And server is responsible to reject bad requests.

Reply:
If values should not be changed by the client, i.e. read only, then they need to be marked "readonly": true.  This seems to be an error on the RP side.  As per the guidelines,

```
Read only parameters of the model properties MUST be labeled as "readOnly": true in OpenAPI(Swagger) spec
```

Reply:
same as other responses.

Reply:
updated